### PR TITLE
feat(wallet): Portfolio Account Filter

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Adds an `Account Filter` to the `Portfolio Overview` page.

Resolves <https://github.com/brave/brave-browser/issues/25587>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://user-images.githubusercontent.com/40611140/192355578-6b55a353-fcf6-4f0f-ae31-7fd61b5c0514.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/account-filter-selector/account-filter-item.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import * as React from 'react'
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/account-filter-selector/account-filter-item.tsx:1]: Missing license

File: components/brave_wallet_ui/components/desktop/account-filter-selector/index.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/account-filter-selector/index.tsx:1]: Can we rename this file to account-filter-selector.tsx ?

File: components/brave_wallet_ui/components/desktop/account-filter-selector/style.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/account-filter-selector/style.ts:1]: Can we rename this file to account-filter-selector.style.tsx ?

File: components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-overview.tsx [hunk 4]
```diff
@@ -119,13 +121,22 @@ export const PortfolioOverview = () => {
...
-    return visibleTokensForSupportedChains.map((asset) => ({
+    return visibleTokensForFilteredAccount.map((asset) => ({
       asset: asset,
-      assetBalance: fullAssetBalance(asset)
+      assetBalance: selectedAccountFilter.id === AllAccountsOption.id
+        ? fullAssetBalance(asset)
+        : getBalance(networkList, selectedAccountFilter, asset)
     }))
-  }, [visibleTokensForSupportedChains, fullAssetBalance])
+  }, [visibleTokensForFilteredAccount, selectedAccountFilter, networkList, fullAssetBalance, getBalance])
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-overview.tsx:139]: getBalance is imported and does not need to be in the decency array

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/account-filter-selector/account-filter-item.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/account-filter-selector/account-filter-item.tsx:6]: Added!

File: components/brave_wallet_ui/components/desktop/account-filter-selector/index.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/account-filter-selector/index.tsx]: Renamed!

File: components/brave_wallet_ui/components/desktop/account-filter-selector/style.ts (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/account-filter-selector/style.ts]: Renamed!

File: components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-overview.tsx (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-overview.tsx]: Removed!

**Review by josheleonard — APPROVED** [review]

**Pull request merged**